docs/enhancements/hypershift-adaptation.md

```diff
@@ -0,0 +1,99 @@
+
+- Once Performance Profile Controller has created the `tuned` object as usual, it will embeded the `tuned` into a `configmap` in the `hosted-control-plane-namespace`.
+  - This `configmap` will have:
+    - label: `hypershift.openshift.io/tuned-config` : `true`
+    - annotation: `hypershift.openshift.io/nodePool` : `NodePool` API name where the `PeformanceProfile` which generate this `tuned` was referenced.
+- This will trigger the reconcile operation in NTO for these kind of objects.
+
+### MachineConfig
+
+The proposal for this output object is to use the way NTO has already put in place to handle its own `MachineConfig` objects as it is.
+
+- Once Performance Profile Controller has created the `MachineConfig` object as usual, it will embeded the object into a `configmap` in the `hosted-control-plane-namespace`.
+- This `configmap` will have:
+  - label: `hypershift.openshift.io/nto-generated-machine-config` : `true`
+  - label: `hypershift.openshift.io/nodePool` : `NodePool` API name where the `PeformanceProfile` which generate this `MachineConfig` was referenced.
+- This will trigger the reconcile operation in NodePool Controller for these objects.
+
+### MCO KubeletConfig
+
+__WIP__
+
+Being this object also handled by MachineConfig Operator (MCO) as `MachineConfig`s the proposal is to handle them in a similar way.
+
+- Once Performance Profile Controller has created the `Kubeletconfig` object as usual, it will embeded the object into a `configmap` in the `hosted-control-plane-namespace`.
+- This `configmap` will have:
+  - label: `hypershift.openshift.io/nto-generated-kubelet-config` : `true`
+  - label: `hypershift.openshift.io/nodePool` : `NodePool` API name where the `PeformanceProfile` which generate this `MachineConfig` was referenced.
+- This will trigger the reconcile operation in NodePool Controller for these objects.
+  - This operation is __NOT__ in place already, so a modification in the NodePool Controller will be needed, but it would be quite similar to the one handling `MachineConfigs` right now.
+
+> __development NOTE__: I am not sure this is the proper way to handle this because I still have some doubts regarding the way NodePool Controller handles `MachineConfig` objects, that's why I have asked @dagrayvid about that
+>
+> Here is the question:
+>
+> - Regarding NTO generated machine configs. I have seen that nodepool controller extract all the machine configurations from the configmaps,
+>  then comprress them and store them into a core.v1.Secret in the hosted-control-plane-namespace ... and I do not understand this operation.
+>  I mean, in the end machineconfig operator should handle this machine configs to apply them in the hosted cluster, right?
+>  I thought there have been some adaptations in the MCO for hypershift to handle this, but I cannot find anything in the repo so any help will be really appreciated.
```

Comment:
As mentioned on Slack, there are a lot of changes in how MachineConfigs are applied in HyperShift. This enhancement will give a better explanation than I can https://github.com/openshift/enhancements/blob/master/enhancements/hypershift/node-lifecycle.md.

To add onto this, if you take a look at the operators running in the control plane namespace, there is no MCO, only the ignition-server. On the hosted cluster nodes, there is no MCD DaemonSet.